services/src/main/java/org/keycloak/social/twitter/TwitterIdentityProvider.java

```diff
@@ -67,14 +70,18 @@ public class TwitterIdentityProvider extends AbstractIdentityProvider<OAuth2Iden
 
     String TWITTER_TOKEN_TYPE="twitter";
 
-
     protected static final Logger logger = Logger.getLogger(TwitterIdentityProvider.class);
 
     private static final String TWITTER_TOKEN = "twitter_token";
-    private static final String TWITTER_TOKENSECRET = "twitter_tokenSecret";
+
+    private final OAuthAuthorization oAuthAuthorization;
 
     public TwitterIdentityProvider(KeycloakSession session, OAuth2IdentityProviderConfig config) {
         super(session, config);
+        VaultStringSecret vaultStringSecret = session.vault().getStringSecret(getConfig().getClientSecret());
```

Comment:
Minor: The VaultStringSecret created here in the constructor is not closed. I guess it should be as it implements `AutoCloseable` and it is closed in other places too. Is it possible to close it right away in the constructor? I

Reply:
Ok, it's done now!